azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub.py

```diff
@@ -35,32 +35,29 @@ def _execute_op(self, op):
             self.auth_provider.on_sas_token_updated_handler = CallableWeakMethod(
                 self, "on_sas_token_updated"
             )
-            self.send_worker_op_down(
-                worker_op=pipeline_ops_iothub.SetIoTHubConnectionArgsOperation(
-                    device_id=self.auth_provider.device_id,
-                    module_id=getattr(self.auth_provider, "module_id", None),
-                    hostname=self.auth_provider.hostname,
-                    gateway_hostname=getattr(self.auth_provider, "gateway_hostname", None),
-                    ca_cert=getattr(self.auth_provider, "ca_cert", None),
-                    sas_token=self.auth_provider.get_current_sas_token(),
-                    callback=op.callback,
-                ),
-                op=op,
+            worker_op = op.spawn_worker_op(
```

Comment:
>spawn_worker_op [](start = 27, length = 15)

Just thinking out loud, does the verb "spawn" imply that this function also sends it down?  It feels like it does, but I don't know why.  Why is this "spawn_worker_op" instead of "create_worker_op"?  What does "spawn" do that "create" doesn't do?

Reply:
i think spawn refers to creating something that will start on a different process. Also verb wise spawn refers to multiple instances and create refers to only 1. 

---
In reply to: [346001572](https://github.com/Azure/azure-iot-sdk-python/pull/392#discussion_r346001572) [](ancestors = 346001572)

Reply:
I don't think spawn implies that it sends it down at all. The reason I chose "spawn" over "create" is that I think it makes the relationship between the original op and the worker op clearer. The original op spawns a worker op, i.e. the worker op comes from the original op. With "create", that isn't as clear.

---
In reply to: [346998748](https://github.com/Azure/azure-iot-sdk-python/pull/392#discussion_r346998748) [](ancestors = 346998748,346001572)